streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -190,16 +313,15 @@ public KafkaStreams(final TopologyBuilder builder, final StreamsConfig config, f
     public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
-        if (state == StreamsState.created) {
-            for (final StreamThread thread : threads) {
+        if (state() == KafkaStreams.State.CREATED) {
```

Comment:
You could access the state variable directly here and in other places throughout. It is volatile so accessing it from multiple threads is ok.